src/dnaio/__init__.py

```diff
@@ -122,21 +131,51 @@ def open(
        A subclass of `SingleEndReader`, `PairedEndReader`, `SingleEndWriter` or
        `PairedEndWriter`.
     """
-    if mode not in ("r", "rb", "w", "a"):
-        raise ValueError("Mode must be 'r', 'rb', 'w' or 'a'")
-    elif interleaved and file2 is not None:
-        raise ValueError("When interleaved is True, file2 must be None")
-    elif interleaved or file2 is not None:
+    if files and (file1 is not None):
+        raise ValueError(
+            "the file1 argument cannot be used together with unnamed files"
```

Comment:
```suggestion
            "The file1 keyword argument cannot be used together with files specified"
            "as positional arguments"
```
(I initially associated "unnamed file" with something filesystem related.)